src/Microsoft.ML.TensorFlow/TensorflowUtils.cs

```diff
@@ -439,15 +439,7 @@ internal static Tensor CastDataAndReturnAsTensor<T>(T[] data, TensorShape tfShap
             else if (typeof(T) == typeof(double))
                 return new Tensor((double[])(object)data, dims, TF_DataType.TF_DOUBLE);
             else if (typeof(T) == typeof(ReadOnlyMemory<char>))
-            {
-                string[] strings = new string[data.Length];
-                for (int i = 0; i < strings.Length; i++)
-                {
-                    strings[i] = ((ReadOnlyMemory<char>)(object)data[i]).ToString();
-                }
-
-                return new Tensor(strings);
-            }
+                return new Tensor(data.Select(x => x.ToString()).ToArray());
```

Comment:
Instead of using Linq here, I would the loop and `new string[data.Length]` code that you had before. It will achieve better performance since you will only allocate a single array.

Reply:
I'm not quite follow here, the size of `data` is fixed, so where will the second memory allocation be when calling `ToArray`

Reply:
Or is that `Select` where the n memory allocation will be

Reply:
Correct - the `Select` will cause an allocation:

https://github.com/dotnet/runtime/blob/4486164506a56200e66cf4cd750fde6b54ab5abd/src/libraries/System.Linq/src/System/Linq/Select.cs#L37